### ISSUE TITLE
change pending_acks in brod_consumer.erl from list to queue?

### ISSUE THREAD

**Issue by theorix** [kafka4beam/brod#162]
I had found buffer and pending was changed from list to queue in brod_producer_buffer.erl,  and buffer_count was used instead of length(buffer).
maybe pending_acks in brod_consumer.erl also need this change?
when I use brod_consumer, the length of pending_acks may more than 20000 some times.

**Comment by zmstone** [comment]
I like this idea! @theorix 
Could you help out with a pr ?

**Comment by theorix** [comment]
glad to do it : )

**Comment by zmstone** [comment]
I have an idea:
most of the time, offsets are sequential, except for compacted topics.
so we can actually make an abstraction for it like this:

```
#pending_acks{ count = 0 :: integer()
               queue_rear = undefined :: One :: offset() | Range :: {offset(), offset()}
               offsets_queue = queue:new() :: queue:queue(One :: offset() | Range :: {offset(), offset()})
```
queue_rear is kept outside offsets_queue to avoid flipping the queue internals when doing queue:in and queue:out_r a lot.

when adding a new offset, we expand the queue_rear range if possible, otherwise push queue_rear to offsets_queue and add it as a One :: offset() to maybe start accumulating a new range.

**Comment by theorix** [comment]
good idea!

**Comment by zmstone** [comment]
fixed in #164
